### PR TITLE
When the https request returns a StatusUnauthorized error code, the tenantID in the url is not actually modified.

### LINKED ISSUE #584: When the https request returns a StatusUnauthorized error code, the tenantID in the url is not actually modified.
**Is this a BUG REPORT or FEATURE REQUEST?**:
> /kind bug

**What happened**:
```
func (k *KeystoneReciver) Recv(url string, method string, body interface{}, output interface{}) error {
	desc := fmt.Sprintf("%s %s", method, url)
	return utils.Retry(2, desc, true, func(retryIdx int, lastErr error) error {
		if retryIdx > 0 {
			err, ok := lastErr.(*HttpError)
			if ok && err.Code == http.StatusUnauthorized {
				k.GetToken()
			} else {
				return lastErr
			}
		}

		headers := HeaderOption{}
		headers[constants.AuthTokenHeader] = k.Auth.TokenID
		return request(url, method, headers, body, output)
	})
}
```
When the https request returns a `StatusUnauthorized `error code, the token is modified, but the tenantID in the url is not actually modified.

**What you expected to happen**:

When `StatusUnauthorized ` occurs, both token and tenantID are modified.

**Environment**:
- Hotpot(release/branch) version: v0.4.0

### PR DESCRIPTION
**Pull request by lasshpanzer**
…enantID in the url is not actually modified.

**What this PR does / why we need it**:

URL contains tenant id. When StatusUnauthorized occurs, not only every token id and tenant id, but also every URL should be modified. I separate the `Config `from `Client `struct. so if unauthorized occurs, every token id, tenant id and URL will be updated.

**Which issue this PR fixes: fixes #584

### CODE REVIEW

File: client/receiver.go [hunk 1]
```diff
@@ -113,9 +113,14 @@ func request(url string, method string, headers HeaderOption, input interface{},
...
-type receiver struct{}
+type receiver struct {
+	Auth *NoAuthOptions
+}
 
-func (*receiver) Recv(url string, method string, input interface{}, output interface{}) error {
+func (r *receiver) Recv(url string, endPoint string, method string, input interface{}, output interface{}, in ...string) error {
+	url := strings.Join([]string{
+		endPoint,
+		urls.GenerateURL(resource, urls.Client, k.Auth.GetTenantId(), in...)}, "/")
```
Comment (leonwanghui) [inline client/receiver.go]: Can you explain where are `resource`, `urls` and `k` come from?

File: client/client.go [hunk 2]
```diff
@@ -26,53 +26,49 @@ const (
...
+// Config is a struct that defines some options for calling the Client.
 type Client struct {
 	*ProfileMgr
 	*DockMgr
 	*PoolMgr
 	*VolumeMgr
 	*VersionMgr
 	*ReplicationMgr
-
-	cfg *Config
```
[the remark this answers is not in the document]
Comment (lasshpanzer) [reply client/client.go]: I resubmitted pr, please review again

File: client/dock.go [hunk 3]
```diff
@@ -50,21 +42,17 @@ func (d *DockMgr) GetDock(dckID string) (*model.DockSpec, error) {
...
-
+	var filter string
 	param, err := processListParam(args)
 	if err != nil {
 		return nil, err
 	}
 
 	if param != "" {
-		url += "?" + param
+		filter += "?" + param
```
[the remark this answers is not in the document]
Comment (lasshpanzer) [reply client/dock.go]: I resubmitted pr, please review again.

File: client/receiver.go (hunk 1 above)
Comment (lasshpanzer) [reply client/receiver.go]: I resubmitted pr, please review again.

**Pull request closed without merging**